main.py

```diff
@@ -15,99 +15,105 @@
+    :param flight_display: Display interface for flight data.
+    :param is_mock: Whether running in simulation mode.
+    """
     try:
-        airbrakes.start()  # Start the IMU and logger processes
-
-        # Setup our flight display
-        # This is what prints the flight data to the console in real time, we only do
-        # it when running the sim because printing a lot of things can significantly slow down
-        # the program
+        # Starts the airbrakes system and display
+        airbrakes.start()
         flight_display.start()
 
-        # This is the main loop that will run until we press Ctrl+C
         while not airbrakes.shutdown_requested:
-            # Update the airbrakes finite state machine
+            # Update the state machine
             airbrakes.update()
 
-            # Stop the sim when the data is exhausted:
-            if args.mock and not airbrakes.imu._data_fetch_process.is_alive():
+            # Stop the simulation when the data is exhausted
+            if is_mock and not airbrakes.imu.is_running:
                 break
+
+    # Handle user interrupt gracefully
     except KeyboardInterrupt:
-        if args.mock:
+        if is_mock:
             flight_display.end_sim_interrupted.set()
-    else:  # This is run if we have landed and the program is not interrupted (see state.py)
-        if args.mock:
-            # Usually the mock sim will stop itself due to data exhaustion, so we will actually
-            # hit the condition above, but if the data isn't exhausted, we will stop it here.
+    else:
+        # Stop the mock simulation naturally if not interrupted
+        if is_mock:
             flight_display.end_sim_natural.set()
     finally:
+        # Stop the display and airbrakes
         flight_display.stop()
         airbrakes.stop()
 
 
+def main(args: argparse.Namespace) -> None:
+    """
+    Sets up and runs the airbrakes system.
+    :param args: Command line arguments for configuration.
+    """
+    sim_time_start = time.time()
+
+    # Create the necessary components
+    servo, imu, logger = create_components(args)
+
+    # Initialize data processing and prediction
+    data_processor = IMUDataProcessor()
+    apogee_predictor = ApogeePredictor()
+
+    # Initialize the airbrakes context and display
+    airbrakes = AirbrakesContext(servo, imu, logger, data_processor, apogee_predictor)
+    flight_display = FlightDisplay(airbrakes, sim_time_start, args)
+
+    # Run the main flight loop
+    run_flight_loop(airbrakes, flight_display, args.mock)
+
+
 if __name__ == "__main__":
-    # Command line args:
+    # Command line args (after these are run, you can press Ctrl+C to exit the program):
     # python main.py -v: Shows the display with much more data
-    # python main.py -m: Runs a simulation on your computer.
+    # python main.py -m: Runs a simulation on your computer
     # python main.py -m -r: Runs a simulation on your computer with the real servo.
```

Comment:
what's the philosophy behind removing the periods here, but not in the next sentence?

Reply:
I forgot this line lol